common/src/main/java/org/apache/uniffle/common/executor/ThreadPoolManager.java

```diff
@@ -0,0 +1,176 @@
+   * @param workQueue the work queue
+   * @param threadFactory the thread factory
+   * @return the registered thread pool
+   */
+  public static ThreadPoolExecutor newThreadPool(
+      String name,
+      int corePoolSize,
+      int maximumPoolSize,
+      long keepAliveTime,
+      TimeUnit unit,
+      BlockingQueue<Runnable> workQueue,
+      ThreadFactory threadFactory) {
+    ThreadPoolExecutor threadPoolExecutor =
+        new ThreadPoolExecutor(
+            corePoolSize, maximumPoolSize, keepAliveTime, unit, workQueue, threadFactory);
+    registerThreadPool(name, corePoolSize, maximumPoolSize, keepAliveTime, threadPoolExecutor);
+    return threadPoolExecutor;
+  }
+
+  /**
+   * Add a thread pool.
+   *
+   * @param name the name of the thread pool
+   * @param corePoolSize the core pool size supplier
+   * @param maximumPoolSize the maximum pool size supplier
+   * @param keepAliveTime the keep alive time supplier
+   * @param unit the unit
+   * @param workQueue the work queue
+   * @param threadFactory the thread factory
+   * @param handler the handler to use when execution is blocked because the thread bounds and queue
+   *     capacities are reached
+   * @return the registered thread pool
+   */
+  public static ThreadPoolExecutor newThreadPool(
+      String name,
+      int corePoolSize,
+      int maximumPoolSize,
+      long keepAliveTime,
+      TimeUnit unit,
+      BlockingQueue<Runnable> workQueue,
+      ThreadFactory threadFactory,
+      RejectedExecutionHandler handler) {
+    ThreadPoolExecutor threadPoolExecutor =
+        new ThreadPoolExecutor(
+            corePoolSize, maximumPoolSize, keepAliveTime, unit, workQueue, threadFactory, handler);
+    registerThreadPool(name, corePoolSize, maximumPoolSize, keepAliveTime, threadPoolExecutor);
+    return threadPoolExecutor;
+  }
+
+  /**
+   * Register a thread pool to THREAD_POOL_MAP.
+   *
+   * @param name the name of the thread pool
+   * @param corePoolSize the core pool size supplier
+   * @param maximumPoolSize the maximum pool size supplier
+   * @param keepAliveTime the keep alive time supplier
+   * @param threadPoolExecutor the thread pool which will be registered
+   */
+  public static void registerThreadPool(
+      String name,
+      int corePoolSize,
+      int maximumPoolSize,
+      long keepAliveTime,
+      ThreadPoolExecutor threadPoolExecutor) {
+    THREAD_POOL_MAP.put(
+        threadPoolExecutor, new MeasurableThreadPoolExecutor(name, threadPoolExecutor));
+    LOG.info(
+        "{} thread pool, core size:{}, max size:{}, keep alive time:{}",
+        name,
+        corePoolSize,
+        maximumPoolSize,
+        keepAliveTime);
+  }
+
+  /**
+   * Unregister the thread pool executor related to the given key.
+   *
+   * @param key the key of thread pool executor to unregister
+   */
+  public static void unregister(Object key) {
+    MeasurableThreadPoolExecutor measurableThreadPoolExecutor = THREAD_POOL_MAP.remove(key);
```

Comment:
I think this is not thread safe. Before real closing this executor, we should ensure the reference of this executor pool is equals to 0.

Reply:
Could you explain a little more?